whois-api/src/test/java/net/ripe/db/whois/api/ApiKeyAuthServerDummy.java

```diff
@@ -48,15 +48,23 @@ public class ApiKeyAuthServerDummy implements Stub {
     public static final String BASIC_AUTH_TEST_TEST_MNT = "dDZsUlpndk9GSXBoamlHd3RDR3VMd3F3OjJDVEdQeDVhbFVFVzRwa1Rrd2FRdGRPNg==";
     public static final String BASIC_AUTH_INVALID_API_KEY = "aDZsUlpndk9GSXBoamlHd3RDR3VMd3F3OjJDVEdQeDVhbFVFVzRwa1Rrd2FRdGRPNg==";
     public static final String BASIC_AUTH_INVALID_SIGNATURE_API_KEY = "TXp1ZzRxRVlpSTVET1dqOXI1Qkp1Y2k4OnZBdzgyRTFCMkZ2dFVyYjB0MDF0Ykt2cg==";
+    public static final String BASIC_AUTH_FULL_ENV_SIGNATURE_API_KEY = "M1RFMEtZWDM1U1pRVUtJQUJNNVZBT0dXOllDZms2a09BaUhYRUl6cnhmTzRTTm9vRQ==";
+    public static final String BASIC_AUTH_NOT_MNT_ENV_SIGNATURE_API_KEY = "U0o5NjkxTlhVS1pGVlpNUVhMUVlCVlJIOnE0dWI4T0xPY3RKR3J5SjhtUHZBQ0tKMQ==";
+    public static final String BASIC_AUTH_WRONG_ENV_SIGNATURE_API_KEY = "UlFNTzVBU05SVU44WkpOUDZUQ1MxUTJROmNvdjRWejdBR2JyNDRjRmZROFBVMVY0Wg==";
+    public static final String BASIC_AUTH_FULL_WRONG_ENV_SIGNATURE_API_KEY = "UabNTzVBU05SVU44WkpOUDZUQ1MxUTJROmNvdjRWejdBR2JyNDRjRmZROFBVMVY0Wg==";
 
     public static final Map<String, OAuthSession> APIKEY_TO_OAUTHSESSION =  Maps.newHashMap();
 
     {
-        APIKEY_TO_OAUTHSESSION.put(BASIC_AUTH_TEST_NO_MNT, new OAuthSession(AUD, "hHZjAbXPtxGxUJCgdwv2ufhY", "test@ripe.net", "8ffe29be-89ef-41c8-ba7f-0e1553a623e5", "profile email"));
+        APIKEY_TO_OAUTHSESSION.put(BASIC_AUTH_TEST_NO_MNT, new OAuthSession(AUD, "hHZjAbXPtxGxUJCgdwv2ufhY", "test@ripe.net", "8ffe29be-89ef-41c8-ba7f-0e1553a623e5", "profile email whois.environment:LOCALHOST.write"));
         APIKEY_TO_OAUTHSESSION.put(BASIC_AUTH_PERSON_NO_MNT, new OAuthSession(AUD, "l6lRZgvOFIphjiGwtCGuLwqw","person@net.net", "906635c2-0405-429a-800b-0602bd716124", null));
-        APIKEY_TO_OAUTHSESSION.put(BASIC_AUTH_PERSON_OWNER_MNT, new OAuthSession(AUD, "l6lRZgvOFIphjiGwtCGuLwqw","person@net.net", "906635c2-0405-429a-800b-0602bd716124", "profile email whois.mntner:OWNER-MNT"));
+        APIKEY_TO_OAUTHSESSION.put(BASIC_AUTH_PERSON_OWNER_MNT, new OAuthSession(AUD, "l6lRZgvOFIphjiGwtCGuLwqw", "person@net.net", "906635c2-0405-429a-800b-0602bd716124", "profile email whois.mntner:OWNER-MNT"));
```

Comment:
Shouldn't environment scope be mandatory or user already had existing keys without environment and this one tests that?

Reply:
At this point we have keys without environment yes. We are just creating keys with environment since we deploy these changes. This is testing the situation when one used has an existing key without env yes.